### PR TITLE
Add golangci-lint

### PR DESCRIPTION
**Pull request by danielmellado**
This commit adds golangci-lint and fixes multiple violations. Some of
them have just been skipped by using nolint directives.

It also adds a new hack/update-deps script to update vendor
dependencies.

Signed-off-by: Daniel Mellado <dmellado@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,6 +101,11 @@ fmt: ## Run go fmt against code.
 	@echo "Running go fmt"
 	go fmt ./...
 
+.PHONY: golangci-lint
+golangci-lint: ## Run golangci-int against code.
```
Comment (donpenney) [inline Makefile:105]: nit: typo in comment

File: Makefile [hunk 2]
```diff
@@ -101,6 +101,11 @@ fmt: ## Run go fmt against code.
...
+	@echo "Running golangci-lint"
+	HOME=/tmp hack/golangci-lint.sh
```
Comment (donpenney) [inline Makefile:107]: There's already a hack/lint.sh and Make target. Any reason to not just add golangci-lint to that existing script? That script could just become the "general purpose" linter tool, possibly growing in the future to include other linters like pylint, shellcheck, bashate, etc.

File: Makefile (hunk 1 above)
Comment (danielmellado) [reply Makefile:105]: Thanks!

File: Makefile (hunk 2 above)
Comment (danielmellado) [reply Makefile:107]: Well, basically I put that out there in order to have a small script per linter. i.e. golint allows you to have the `-set-exit-status param`which is used in that bash script but it wouldn't apply to golangci-lint nor to anyone else. What we could do, though, is to aggregate such scrips calls into one while keeping them separate into files.

File: controllers/precacheFsm.go [hunk 3]
```diff
@@ -200,11 +201,14 @@ func (r *ClusterGroupUpgradeReconciler) handleNotStarted(ctx context.Context,
 
 // handlePreparing handles conditions in PrecacheStatePreparingToStart
 // returns: error
+//nolint:unparam
 func (r *ClusterGroupUpgradeReconciler) handlePreparing(ctx context.Context,
 	clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade,
 	cluster string) (string, error) {
 
+	//nolint:ineffassign
 	currentState, nextState := PrecacheStatePreparingToStart, PrecacheStatePreparingToStart
```
Comment (vitus133) [inline controllers/precacheFsm.go:210]: I wonder why do I have ineffectual assign here

Comment (vitus133) [inline controllers/precacheFsm.go:210]: I wonder why do I have ineffectual assign here

**Pull request merged**